taigit/src/libraries/Taiga.ts

```diff
@@ -78,4 +78,124 @@ task_history(taskId : number) : Promise<Object> {
     }
 
     return output;
-}
\ No newline at end of file
+}
+
+/**
+ * @summary This call assess a task based on task_assessment_state_trans based on task Id
+ * @param taskId the ID for the task to assess task
+ * @returns  whether the task's state transition is valid,task status transition, date based on task Id
+ * * {
+ * Current in Use
+ *      state_trans_invalid: Boolean       // true, the task state transition is not valid; false, the task is valid.
+ *      date : number,         // Date and time of history entry in milliseconds since epoch
+ *      user : object,         // Taiga User Object
+ *      status_trans:entry.String[],//Status transition array
+ *
+ *  Can be added
+ *      diff_types : String[], // Names of entries in diff_values //diff_values : entry.values_diff,
+ *      diff_values : Object   // Values of the changes listed in diff_types //diff_values : entry.values_diff,
+ * }
+ */
+export async function
+task_assessment_state_trans(taskId : number) : Promise<Object> {
+let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
+    //test case :https://api.taiga.io/api/v1/history/task/2577741
+let output : Array<Object> = [];
+for(let entry of data) {
+    let new_entry = {
+        state_trans_invalid: false,
+        date : new Date(entry.created_at).getTime(),
+        user : entry.user,
+        status_trans:entry.values_diff.status
+    }
+
+    //Only three status transistion is valid
```

Comment:
From an initial review, I think you should consider the instances where students make incorrect transitions. There are times where items are moved by accident (ie things are fixed within a certain timeframe), or that tasks must be returned to certain stages to accommodate things.
This analysis was intended to not only provide a way to assess a student's process (you seem to have accomplished this), but also show the progression of a task for frontend to display. I am not sure your current code allows for this (this comment is not meant to change your development but rather to inspire more considerations for your work)

Reply:
I think about the situation when I write the API. It can be an improvement in the future sprint.